Fatras/include/ActsFatras/Digitization/detail/ParametersCast.hpp

```diff
@@ -0,0 +1,75 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2016-2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Utilities/Definitions.hpp"
+#include "Acts/Utilities/Helpers.hpp"
+#include "Acts/Utilities/ParameterDefinitions.hpp"
+
+namespace ActsFatras {
+namespace detail {
+
+/// Cast function definition for bound parameters
+using BoundCastFunction =
+    std::function<double(const Acts::Vector2D&, const Acts::Vector3D&, double)>;
+
+/// Cast operator to provide local 0
+struct BoundLoc0Cast {
+  double operator()(const Acts::Vector2D& lhit,
+                    const Acts::Vector3D& /*ignored*/, double /*ignored*/) {
+    return lhit[Acts::eBoundLoc0];
```

Comment:
The `eBoundLoc1` access is only well-defined if used for a `BoundVector`. To get the first local position from a position vector you should use `ePos0`.

Reply:
After discussion with @XiaocongAi  - this file will go, it's quite some complex code for a 2-times saving of trigonometric functions (which probably could even be done by two if statements).

Reply:
I think it should be `eBoundLoc1` since the input is a local hit.

Reply:
Irrelevant, the file is gone.